src/ui/buttons/buttonDesigns/switch-logo-and-show-label-text.jsx

```diff
@@ -0,0 +1,144 @@
+import { BUTTON_SIZE_STYLE } from '../config';
+
+import type { ContentOptions } from './types';
+
+export const SWITCH_LOGO_AND_SHOW_LABEL_CONFIG = {
+    min:                            BUTTON_SIZE_STYLE.tiny.minWidth,
+    smallMax:                       BUTTON_SIZE_STYLE.small.maxWidth,
+    mediumMax:                      BUTTON_SIZE_STYLE.medium.maxWidth,
+    cssClasses: {
+        DOM_READY:                  CLASS.DOM_READY,
+        ANIMATION_CONTAINER:        'personalized-design-container',
+        PAYPAL_LOGO:                LOGO_CLASS.LOGO,
+        ANIMATION_LABEL_CONTAINER: 'personalized-label-container',
+        PAYPAL_BUTTON_LABEL:        CLASS.BUTTON_LABEL
+    }
+};
+
+type AnimationProps = {|
+    paypalLabelContainerElement : Object,
+    labelFontSize : number
+|};
+
+export function SwitchLogoAndShowLabelTextComponent({ designLabelText, logoColor } : ContentOptions) : ChildType {
+    // experimentName must match elmo experiment name
+    const ANIMATION_CONTAINER = SWITCH_LOGO_AND_SHOW_LABEL_CONFIG.cssClasses.ANIMATION_CONTAINER;
+    const ANIMATION_LABEL = SWITCH_LOGO_AND_SHOW_LABEL_CONFIG.cssClasses.ANIMATION_LABEL_CONTAINER;
+    return (
+        <div class={ ANIMATION_LABEL } data-design-experiment='104519'>
+            <PPLogo logoColor={ logoColor } />
+            <span>{designLabelText}</span>
+            <style innerHTML={ `
+                .${ CLASS.DOM_READY } .${ ANIMATION_CONTAINER } img.${ LOGO_CLASS.LOGO }-pp{
+                    position: relative;
+                    opacity:0;
+                }
+                .${ ANIMATION_CONTAINER } .${ ANIMATION_LABEL } {
+                    opacity: 0; 
+                    color: #142C8E;
+                    font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
+                }
+            ` } />
+        </div>
+    );
+}
+
+// Returns label container if the button sizes match
+export const getSwitchLogoAndShowLabelProps = function(document : Object, configuration : Object) : AnimationProps | null {
+    let labelFontSize = 8;
+    const { ANIMATION_CONTAINER, ANIMATION_LABEL_CONTAINER, PAYPAL_BUTTON_LABEL } = configuration.cssClasses;
+    const { min, smallMax, mediumMax } = configuration;
+    // get the animation main container to force specificity( in css ) and make sure we are running the right animation
+    const animationContainer = (document && document.querySelector(`.${ ANIMATION_CONTAINER }`)) || null;
+    if (!animationContainer) {
+        return null;
+    }
+
+    // return null if animation should not be played for the button size
+    const animationContainerWidth = animationContainer.offsetWidth;
+    
+    if (animationContainerWidth < min || animationContainerWidth >= mediumMax) {
+        // remove label element from dom
+        animationContainer.querySelector(`.${ ANIMATION_LABEL_CONTAINER }`).remove();
+        return null;
+    }
+
+    if (animationContainerWidth >= smallMax) {
```

Comment:
Can this device width logic be a ternary instead of using a let and reassigning

Reply:
I agree